### PR TITLE
remove gcc8 from CI

### PR DESCRIPTION
**Pull request by vhvb1989**

### CODE REVIEW

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml
Comment (ahsonkhan) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:25]: What caused all these new whitespace changes to this file? Will someone else editing this cause another set of whitespace changes?

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml
Comment (ahsonkhan) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:36]: If you ignore the whitespace changes in this PR, this is the substantive change here, which is removing the linux gcc8 leg from the matrix.

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml
Comment (ahsonkhan) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:32]: Now all the properties align correctly. Good change.

@danieljurek Isn't yml impacted by indentation/spacing? Would outdenting some properties cause correctness issues?

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml
Comment (ahsonkhan) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml:36]: @vhvb1989 can you please file an issue to track why the gcc8 leg is failing. I am assuming removing it from CI is a temporary measure to unblock the current PRs.

Do we plan to investigate and bring this leg back?

**Review by ahsonkhan — APPROVED** [review]
I would hold off on merging until someone from the engsys side has also gotten a chance to review and approve this.

**Comment by Jinming-Hu** [comment]
why do you want to remove that?

**Review by Jinming-Hu — APPROVED** [review]

**Comment by vhvb1989** [comment]
> why do you want to remove that?
> 
> Edit:
> I noticed all CI failed with GCC 8. Do we know the root cause? Is there something wrong with Azure Pipelines?

Still waiting on @danieljurek to review,  the pipelines went down  near to the end of the day because the Linux nodes don't have gcc8 any more.
If you need to merge something and can't wait for tomorrow, you can either override the check enforcer or merge this or to unblock.
Otherwise let's wait for eng sys to review and fix

**Comment by ahsonkhan** [comment]
@vhvb1989 can we close this now that https://github.com/Azure/azure-sdk-for-cpp/pull/2012 is merged?

**Pull request closed without merging**
